tony/tony.sh

```diff
@@ -0,0 +1,163 @@
+#!/bin/bash
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This script installs TonY on a master node within a Google Cloud Dataproc cluster.
+
+set -x -e
+
+readonly TONY_INSTALL_FOLDER='/usr/local/src'
+readonly TONY_SAMPLES_FOLDER="${TONY_INSTALL_FOLDER}"'/TonY-samples'
+readonly TENSORFLOW_VERSION='1.9'
+readonly PYTORCH_VERSION='0.4.1'
+
+# TonY configurations
+readonly NUM_WORKERS=2
+readonly WORKER_MEMORY='4g'
+readonly PS_MEMORY='4g'
+readonly NUM_GPUS=0     # Not supported in Dataproc
+
+
+function err() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@" >&2
+  return 1
+}
+
+function download_and_build_tony() {
+  # Download TonY latest distribution.
+  cd "${TONY_INSTALL_FOLDER}"
+  git clone https://github.com/linkedin/TonY.git
+  cd TonY
+  # Build TonY without tests.
+  ./gradlew build -x test
+  return 0
+}
+
+function install_samples() {
+
+  # Create samples directory structure.
+  mkdir -p "${TONY_SAMPLES_FOLDER}"/deps
+  # Create TensorFlow directory
+  mkdir -p "${TONY_SAMPLES_FOLDER}"/jobs/TFJob/src
+  # Create PyTorch directory
+  mkdir -p "${TONY_SAMPLES_FOLDER}"/jobs/PTJob/src
+
+  # Copy Jar file.
+  cp "${TONY_INSTALL_FOLDER}"/TonY/tony-cli/build/libs/tony-cli-0.1.5-all.jar "${TONY_SAMPLES_FOLDER}"
+
+  # Collect Metadata
+  num_gpus="$(/usr/share/google/get_metadata_value attributes/num_gpus)" || num_gpus="${NUM_GPUS}"
```

Comment:
Best to set this to NUM_GPUS (`0`) for now if Dataproc doesn't support GPUs. If a user sets something greater than 0, I think the TonY job may fail when it tries to ask YARN for, say, 1 GPU, and it doesn't have any (Not 100% sure about this, though. YARN may just ignore the GPU request.).